Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,371 @@ import UIKit
+            headerView.frame.size.height = 1
+            return headerView
+        }
+    }
+
+    private func getInputOutputSectionHeaderView(section: Int,
+                                                 title: String,
+                                                 isCollapsed: Bool) -> UIView {
+        let headerView = UIView.init(frame: CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: 48))
+        headerView.backgroundColor = UIColor.white
+        headerView.frame.size.height = 48
+
+        let headerSeparator = UIView.init(frame: CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: 1))
+        headerSeparator.backgroundColor = UIColor.appColors.gray
+        headerView.addSubview(headerSeparator)
+
+        let headerLabel = UILabel.init(frame: CGRect(x: 16, y: 1, width: self.view.frame.size.width - 56, height: 47))
+        headerLabel.textColor = UIColor.appColors.bluishGray
+        headerLabel.font = UIFont(name: "SourceSansPro-Regular", size: 14)
+        headerLabel.numberOfLines = 1
+        headerLabel.text = title
+        headerView.addSubview(headerLabel)
+
+        let arrowImageView = UIImageView.init(frame: CGRect(x: self.view.frame.size.width - 40, y: 12, width: 24, height: 24))
+        let arrowImage = UIImage(named: "ic_collapse")
+        if !isCollapsed {
+            arrowImageView.image = arrowImage
+        } else {
+            arrowImageView.image = UIImage(cgImage: (arrowImage?.cgImage!)!, scale: CGFloat(1.0), orientation: .downMirrored)
+        }
+        headerView.addSubview(arrowImageView)
+
+        let tapGestureRecognizer = UITapGestureRecognizer(
+            target: self,
+            action: #selector(inputOutputSectionHeaderViewTapped(_:))
+        )
+        headerView.tag = section
+        headerView.addGestureRecognizer(tapGestureRecognizer)
+
         return headerView
     }
-    
+
+    @objc func inputOutputSectionHeaderViewTapped(_ sender: UITapGestureRecognizer?) {
+        guard let section = sender?.view?.tag else { return }
+
+        if section == 1 {
+            self.isTxInputsCollapsed = !self.isTxInputsCollapsed
+            self.transactionDetailsTable.reloadData()
+        } else if section == 2 {
+            self.isTxOutputsCollapsed = !self.isTxOutputsCollapsed
+            self.transactionDetailsTable.reloadData()
+        }
+    }
+
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         switch indexPath.section {
         case 0:
             let cell = tableView.dequeueReusableCell(withIdentifier: "TransactionDetailCell") as! TransactionDetailCell
-            cell.txnDetails = self.generalTxDetails[indexPath.row]
+            cell.txDetail = self.generalTxDetails[indexPath.row]
+            cell.onTxDetailValueCopied = { copiedDetail in
+                Utils.showBanner(parentVC: self, type: .success, text: String(format: LocalizedStrings.sgCopied, copiedDetail))
```

Comment:
#592 (now merged) makes it possible to show a banner in a dialog using
`Utils.showBanner (in: self.view.subviews.first!, ...)`

Can do same after rebasing your PR.